### PR TITLE
feat(fx-2367) adds show2 location map

### PR DESCRIPTION
**Pull request by dzucconi**
The type of this PR is: Feature

This PR resolves [FX-2367]

### Description

* Refactors some component fragment spaghetti
* Adds location map

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[FX-2367]: https://artsyproduct.atlassian.net/browse/FX-2367

### CODE REVIEW

File: src/lib/Components/HoursCollapsible.tsx [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
+import { createFragmentContainer, graphql } from "react-relay"
 
 /**
  * FIXME: RN Collapse implementation diverges from web and we're using props
  * unavailable in exported type, see https://github.com/artsy/palette/issues/137
  */
 const Collapse = _Collapse as React.ComponentClass<any>
 
-type OpeningHours = LocationMap_location["openingHours"]
+type OpeningHours = HoursCollapsible_location["openingHours"]
```
Comment (dzucconi) [inline src/lib/Components/HoursCollapsible.tsx:16]: I assume this isn't super common but this is *super annoying* when it does happen.

File: src/lib/Components/LocationMap/index.tsx [hunk 2]
```diff
@@ -21,15 +21,8 @@ const MapWrapper = styled(Flex)`
...
 
-export enum PartnerType {
-  gallery = "Gallery",
-  museum = "Museum",
-  fair = "Fair",
-}
-
 interface Props {
   location: LocationMap_location
-  partnerType: PartnerType
```
Comment (dzucconi) [inline src/lib/Components/LocationMap/index.tsx:32]: This wasn't being used

File: src/lib/Components/LocationMap/index.tsx [hunk 3]
```diff
@@ -250,29 +243,13 @@ export const LocationMapContainer = createFragmentContainer(LocationMap, {
...
       summary
       coordinates {
         lat
         lng
       }
-      day_schedules: daySchedules {
-        start_time: startTime
-        end_time: endTime
-        day_of_week: dayOfWeek
-      }
```
Comment (dzucconi) [inline src/lib/Components/LocationMap/index.tsx:264]: Not used in anything

File: src/lib/Components/LocationMap/index.tsx [hunk 4]
```diff
@@ -250,29 +243,13 @@ export const LocationMapContainer = createFragmentContainer(LocationMap, {
...
-        ... on OpeningHoursArray {
-          schedules {
-            days
-            hours
-          }
-        }
-        ... on OpeningHoursText {
-          text
-        }
-      }
```
Comment (dzucconi) [inline src/lib/Components/LocationMap/index.tsx:275]: Was being fetched for an unrelated component (HoursCollapsible)

**Review by brainbicycle — APPROVED** [review]

**Pull request merged**